planner/core/expression_rewriter.go

```diff
@@ -326,6 +326,23 @@ func (er *expressionRewriter) handleWindowFunction(v *ast.WindowFuncExpr) (ast.N
 	return v, true
 }
 
+func (er *expressionRewriter) buildSemiApplyFromEqualSubq(np LogicalPlan, l, r expression.Expression, not bool) {
+	var condition expression.Expression
+	if rCol, ok := r.(*expression.Column); ok && (er.asScalar || not) {
+		rCol.InOperand = true
+		// If both input columns of `!= all / = any` expression are not null, we can treat the expression
+		// as normal column equal condition.
+		if lCol, ok := l.(*expression.Column); ok && mysql.HasNotNullFlag(lCol.GetType().Flag) && mysql.HasNotNullFlag(rCol.GetType().Flag) {
```

Comment:
how about:
```go
lCol, ok := l.(*expression.Column);
rCol.InOperand = ok && mysql.HasNotNullFlag(lCol.GetType().Flag) && mysql.HasNotNullFlag(rCol.GetType().Flag)
```

Reply:
It should be like this?
```golang
lCol, ok := l.(*expression.Column);
rCol.InOperand = !(ok && mysql.HasNotNullFlag(lCol.GetType().Flag) && mysql.HasNotNullFlag(rCol.GetType().Flag))
```
this looks not as readable as the current version?